src/app/components/client/toolbar/UpsellBadge.tsx

```diff
@@ -66,15 +68,20 @@ export function UpsellButton(
       <Button {...triggerProps} variant="primary" small>
         {props.label}
       </Button>
-      {dialogState.isOpen && (
-        <UpsellDialog
-          {...overlayProps}
-          state={dialogState}
-          monthlySubscriptionUrl={props.monthlySubscriptionUrl}
-          yearlySubscriptionUrl={props.yearlySubscriptionUrl}
-          subscriptionBillingAmount={props.subscriptionBillingAmount}
-        />
-      )}
+      {/* Test already handled in Dashboard tests */
+      /* c8 ignore next 4 */}
+      {dialogState.isOpen &&
+        (props.enabledFeatureFlags.includes("DisableOneRepScans") ? (
```

Comment:
@codemist The newly introduced condition now says: “If the dialog is open AND the feature flag `DisableOneRepScans` is enabled, show the `WaitlistDialog`. Otherwise, show the `UpsellDialog`”. Doesn’t that mean that if `dialogState.isOpen === false` we show the `UpsellDialog`, which seems like it is not the intended behaviour?